src/main/java/net/fabricmc/loader/discovery/ModResolver.java

```diff
@@ -157,6 +157,14 @@ private static IVecInt toVecInt(IntStream stream) {
 				for (ModCandidate mod : candidateIntMap.keySet()) {
 					int modClauseId = candidateIntMap.get(mod);
 
+					// When a mod provides an alias id, another mod with same id as the alias must NOT be present.
+
+					for (String provided : mod.getInfo().getProvides()) {
+						if (mandatoryMods.contains(provided)) {
```

Comment:
This isnt right, `mandatoryMods` are only mods from the mods dir, and not mods provided via JIJ.

There already seems to be an atempt at handling this here: https://github.com/FabricMC/fabric-loader/blob/master/src/main/java/net/fabricmc/loader/FabricLoader.java#L342-L344